cpp/src/fil/common.cuh

```diff
@@ -148,7 +157,7 @@ struct predict_params : shmem_size_params {
   float* preds;
   const float* data;
   // number of data rows (instances) to predict on
-  size_t num_rows;
+  long long num_rows;
```

Comment:
it's pretty much impossible num_rows will not fit into the new type (see RAM sizes on modern servers); there were some insidious bugs related to unsigned underflow and it's not worth storing in `size_t` only to convert to `long long` for all useful operations.